tests/mdev_configuration_allocation_test.go

```diff
@@ -118,14 +118,16 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 		var desiredMdevTypeName string = "nvidia-222"
 		var expectedInstancesNum int = 16
 		var config v1.KubeVirtConfiguration
+		var originalConfig v1.KubeVirtConfiguration
 		var mdevTestLabel = "mdevTestLabel1"
 
 		BeforeEach(func() {
 			kv := util.GetCurrentKv(virtClient)
 
 			By("Creating a configuration for mediated devices")
 			config = kv.Spec.Configuration
-			config.DeveloperConfiguration.FeatureGates = []string{virtconfig.GPUGate}
+			originalConfig = *config.DeepCopy()
```

Comment:
It will be cleaner to use `DeferCleanup` in this case.

There are several options to it, depending on the called func.
e.g. `DeferCleanup(func() {tests.UpdateKubeVirtConfigValueAndWait(originalConfig)})`

Reply:
Thanks @EdDev, you are definitely right!

Reply:
@fossedihelm @EdDev Hi, we already have the [`resetToDefaultConfig()`](https://github.com/kubevirt/kubevirt/blob/5e8b0f9f3f0ded79b7ebdf327fea2d61b9f7659f/tests/utils.go#L2051) function which is being called as part of the `TestCleanup()`, in the top-level [`AfterEach`](https://github.com/kubevirt/kubevirt/blob/5e8b0f9f3f0ded79b7ebdf327fea2d61b9f7659f/tests/tests_suite_test.go#L97) 
So technically I think the defer is not needed.

Reply:
I actually think that an explicit revert is much better then the global revert.
We cannot enforce it now, but ideally the global one should have been just an assertion to see it is "clean", not to do a revert.

This assured each test is responsible to return things back. And it makes sense, because only individual tests know what they need to revert back. It is extremely wasteful to perform many reverts in vain, when most tests do not need it at all.

Reply:
Thanks @enp0s3. I forgot that there was the `TestCleanup()`. @EdDev I think that you have great arguments, but currently,  in most of the cases we are not explicitly revert it. Probably this requires a wider discussion and a sort of strategy to migrate toward it. Also, I just notice that the revert is already applied here https://github.com/kubevirt/kubevirt/blob/835683a546d123774c99787c55aa6cdf28d5fabd/tests/mdev_configuration_allocation_test.go#L155.